src/lib.rs

```diff
@@ -4221,6 +4255,291 @@ fn map_darwin_target_from_rust_to_compiler_architecture(target: &Target) -> &str
+        if flag.starts_with("-Z") {
+            self.set_unstable_rustc_flag(flag);
+        } else {
+            self.set_stable_rustc_flag(flag);
+        }
+    }
+
+    fn set_stable_rustc_flag(&mut self, flag: &str) {
+        let (flag, value) = if let Some((flag, value)) = flag.split_once('=') {
+            (flag, Some(value.to_owned()))
+        } else {
+            (flag, None)
+        };
+
+        match flag {
+            "-Ccode-model" => self.code_model = value,
+            "-Cno-vectorize-loops" => self.no_vectorize_loops = true,
+            "-Cno-vectorize-slp" => self.no_vectorize_slp = true,
+            "-Cprofile-generate" => self.profile_generate = value,
+            "-Cprofile-use" => self.profile_use = value,
+            "-Ccontrol-flow-guard" => self.control_flow_guard = value.or(Some("true".into())),
+            "-Clto" => self.lto = value.or(Some("true".into())),
+            "-Crelocation-model" => self.relocation_model = value,
+            "-Cembed-bitcode" => {
+                self.embed_bitcode = value.map_or(Some(true), |val| match val.as_str() {
+                    "y" | "yes" | "on" | "true" => Some(true),
+                    "n" | "no" | "off" | "false" => Some(false),
+                    _ => None,
+                });
+            }
+            "-Cforce-frame-pointers" => {
+                self.force_frame_pointers = value.map_or(Some(true), |val| match val.as_str() {
+                    "y" | "yes" | "on" | "true" => Some(true),
+                    "n" | "no" | "off" | "false" => Some(false),
+                    _ => None,
+                });
+            }
+            "-Clink-dead-code" => {
+                self.link_dead_code = value.map_or(Some(true), |val| match val.as_str() {
+                    "y" | "yes" | "on" | "true" => Some(true),
+                    "n" | "no" | "off" | "false" => Some(false),
+                    _ => None,
+                })
+            }
+            "-Cno-redzone" => {
+                self.no_redzone = value.map_or(Some(true), |val| match val.as_str() {
+                    "y" | "yes" | "on" | "true" => Some(true),
+                    "n" | "no" | "off" | "false" => Some(false),
+                    _ => None,
+                });
+            }
+            "-Csoft-float" => {
+                self.soft_float = value.map_or(Some(true), |val| match val.as_str() {
+                    "y" | "yes" | "on" | "true" => Some(true),
+                    "n" | "no" | "off" | "false" => Some(false),
+                    _ => None,
+                });
+            }
+            _ => {}
+        }
+    }
+
+    fn set_unstable_rustc_flag(&mut self, flag: &str) {
+        let (flag, value) = if let Some((flag, value)) = flag.split_once('=') {
+            (flag, Some(value.to_owned()))
+        } else {
+            (flag, None)
+        };
+
+        match flag {
+            "-Zbranch-protection" => self.branch_protection = value,
```

Comment:
Actually, how about we match this one on `"-Zbranch-protection" | "-Cbranch-protection"` straight away? That way the crate won't need to be updated when the option gets stabilised

Reply:
Yeah if you merge the unstable/stable flags, then I think matching it like this makes sense